crates/autopilot/src/solvable_orders.rs

```diff
@@ -193,9 +196,22 @@ impl SolvableOrdersCache {
         let removed = counter.checkpoint("dust_order", &orders);
         filtered_order_events.extend(removed);
 
+        let cow_amm_tokens = cow_amms
+            .iter()
+            .flat_map(|cow_amm| cow_amm.traded_tokens())
+            .unique()
+            .cloned()
+            .collect::<Vec<_>>();
+
         // create auction
-        let (orders, mut prices) =
-            get_orders_with_native_prices(orders, &self.native_price_estimator, self.metrics);
+        let (orders, mut prices) = get_orders_with_native_prices(
+            orders,
+            &self.native_price_estimator,
+            self.metrics,
+            cow_amm_tokens,
+            self.native_price_timeout,
+        )
+        .await;
```

Comment:
My other comment is resolved, but I still don't see this being surrounded by a metric.